android/quest/src/main/java/org/smartregister/fhircore/quest/QuestApplication.kt

```diff
@@ -33,6 +39,13 @@ class QuestApplication : Application(), DataCaptureConfig.Provider {
     if (BuildConfig.DEBUG) {
       Timber.plant(Timber.DebugTree())
     }
+
+    CoroutineScope(SupervisorJob() + Dispatchers.Default).launch {
```

Comment:
Any reason for using a `SupervisorJob` here? do you intend to handle the coroutine exceptions independently? I actually don't think using a `SupervisorJob` here is necessary as we only have one task.

Reply:
**SupervisorJob**

> Similar to a regular Job with the only exception that its children can fail independently of each other. A child’s failure or cancellation does not result in the supervisor’s job failing or affecting its other children, therefore a supervisor can create a unique policy for dealing with its children’s failures.